samcli/hook_packages/terraform/hooks/prepare/hook.py

```diff
@@ -77,6 +77,11 @@
 TERRAFORM_BUILD_SCRIPT = "copy_terraform_built_artifacts.py"
 TF_BACKEND_OVERRIDE_FILENAME = "z_samcli_backend_override"
 
+HOOK_METADATA_KEY = "AWS::SAM::Hook"
+TERRAFORM_HOOK_METADATA = {
+    "HookPackageId": "terraform",
```

Comment:
Should we update this as we are changing from `--hook-package-id` to `--hook-name`? (https://github.com/aws/aws-sam-cli/pull/4382/files)

```suggestion
    "HookName": "terraform",
```

Reply:
we did not change the collected telemetry .. to do that we need to update our telemetry service .. I thought it is Ok to leave it as it is as it is something internal and does not affect our customers

Reply:
we discussed this outside here, but yes we can do this change independently of the telemetry one (that would be a separate change, if we decided to do that, but we can also keep that internally as Mohamed mentions). 

I just pushed the change to rename